test/hotspot/jtreg/vmTestbase/nsk/jdb/interrupt/interrupt001/interrupt001a.java

```diff
@@ -122,8 +122,8 @@ public int runIt(String args[], PrintStream out) {
 
         long waitTime = argumentHandler.getWaitTime() * 60 * 1000;
         long startTime = System.currentTimeMillis();
-        while (notInterrupted.get() > 0 && System.currentTimeMillis() - startTime <= waitTime) {
-            synchronized (waitnotify) {
+        synchronized (waitnotify) {
+            while (notInterrupted > 0 && System.currentTimeMillis() - startTime <= waitTime) {
```

Comment:
@plummercj I'd use a monotonic clock like `System.nanoTime()` for things like this since the system clock can be changed to anything at any time.

Reply:
> I'd use a monotonic clock like `System.nanoTime()` for things like this since the system clock can be changed to anything at any time.

We have 100's of uses of `System.currentTimeMillis()` in our tests. I think changing them should be handled by a separate effort.